k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/ImapFolder.java

```diff
@@ -860,6 +861,85 @@ public void fetchPart(Message message, Part part, MessageRetrievalListener<Messa
         }
     }
 
+    @Override
+    public void fetchPartWithProgressCallback(Message message, Part part, MessageRetrievalListener<Message> listener, AttachmentProgressCallback progressCallback) throws MessagingException {
```

Comment:
This is a lot of new code in one large function. Is it a slightly altered version of another function? If so, can we reduce the code duplication please.